src/libraries/Common/tests/System/Net/StressTests/run-docker-compose.sh

```diff
@@ -86,15 +92,17 @@ if [[ "$buildcurrentlibraries" -eq 1 ]]; then
     fi
 fi
 
-build_args="--build-arg VERSION=$version --build-arg CONFIGURATION=$configuration"
-if [[ -n "$imagename" ]]; then
-    build_args="$build_args --build-arg SDK_BASE_IMAGE=$imagename"
-fi
-
 compose_file="$projectdir/docker-compose.yml"
 
-if ! docker-compose --file "$compose_file" build $build_args; then
-    exit $?
+if [[ "$nobuild" -eq 0 ]]; then
+    build_args="--build-arg VERSION=$version --build-arg CONFIGURATION=$configuration"
+    if [[ -n "$imagename" ]]; then
+        build_args="$build_args --build-arg SDK_BASE_IMAGE=$imagename"
+    fi
+
+    if ! docker-compose --file "$compose_file" build $build_args; then
```

Comment:
```suggestion
    if ! docker compose --file "$compose_file" build $build_args; then
```
I was able to run this on both WSL2 and Ubuntu, but only with `docker compose` for both building and running. @rzikm did you experience problems with that?

Reply:
And it did not work on my machine and CI :D 

What OS version (and docker versions) are you running? On my freshly updated Ubuntu 22.04 LTS it does not recognize `docker compose`

```
rzikm@rzikm-Virtual-Machine:/source/dotnet/runtime/src/libraries/System.Net.Http/tests/StressTests/HttpStress$ docker compose up
docker: 'compose' is not a docker command.
See 'docker --help'
rzikm@rzikm-Virtual-Machine:/source/dotnet/runtime/src/libraries/System.Net.Http/tests/StressTests/HttpStress$ docker compose build
docker: 'compose' is not a docker command.
See 'docker --help'
rzikm@rzikm-Virtual-Machine:/source/dotnet/runtime/src/libraries/System.Net.Http/tests/StressTests/HttpStress$ docker --version
Docker version 24.0.7, build 24.0.7-0ubuntu2~22.04.1
rzikm@rzikm-Virtual-Machine:/source/dotnet/runtime/src/libraries/System.Net.Http/tests/StressTests/HttpStress$ docker-compose --version
docker-compose version 1.29.2, build unknown
rzikm@rzikm-Virtual-Machine:/source/dotnet/runtime/src/libraries/System.Net.Http/tests/StressTests/HttpStress$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.5 LTS"
```

Reply:
My setup:
- VM: `Ubuntu 22.04.5 LTS` and docker `27.3.1`
- WSL2: `Ubuntu 22.04.3 LTS`, docker `20.10.22`

> on my machine

Does `sudo apt-get install docker-compose-plugin` help?

> and CI

That's weird bc the runner command is `docker compose`, IMO we should harmonize the script.

Reply:
> Does sudo apt-get install docker-compose-plugin help?

there seems to be no package `docker-compose-plugin` in default feeds

it probably depends where you installed the docker from, the 24.0.7 comes from official ubuntu jammy feed

```
rzikm@rzikm-Virtual-Machine:~$ apt-cache madison docker.io
 docker.io | 24.0.7-0ubuntu2~22.04.1 | http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages
 docker.io | 20.10.21-0ubuntu1~22.04.3 | http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages
 docker.io | 20.10.12-0ubuntu4 | http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages
```

And since this likely is the version running in the build image we are using, I think we need to keep `docker-compose` on Linux for now.

Reply:
The `apt-cache madison docker-compose-plugin` lists a bunch of packages from the jammy feed for me:

<details>

```
anton@PRG-NET-TOWER-01:~$ apt-cache madison docker.io
    docker.io | 24.0.7-0ubuntu2~22.04.1 | http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages
    docker.io | 20.10.21-0ubuntu1~22.04.3 | http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages
    docker.io | 20.10.12-0ubuntu4 | http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages

anton@PRG-NET-TOWER-01:~$ apt-cache madison docker-compose-plugin
    docker-compose-plugin | 2.29.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.29.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.29.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.28.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.27.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.27.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.26.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.25.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.24.7-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.24.6-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.24.5-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.24.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.24.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.21.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.20.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.19.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.18.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.17.3-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.17.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.16.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.15.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.14.1~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.12.2~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.12.0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.11.2~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.10.2~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.6.0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.5.0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
    docker-compose-plugin | 2.3.3~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
```

</details>

Also the line I commented on is about the `$buildonly` branch which is currently used for local runs only. In this PR the CI actually *did run* `docker compose` thx to its' usage in the `$nobuild` branch below, and the runs are passing. `docker-compose` doesn't work on my machines, which seems to be an inherent issue with the obsolete tool, not a packaging problem which IMO can be fixed or worked around. Unless you are really sure that `docker compose` may bite more of our team members when doing local runs than `docker-compose`, I think we should proceed with the change, and harmonize all scripts around `docker compose`.

Reply:
> but my Ubuntu VM I still have to invoke `docker-compose` (it does not recognize `docker compose` at all, version 24.0.7)

Are you sure you are experiencing same problems locally and on the CI?

The line this discussion refers to lives in the `$buildonly` branch which is currently used for local manual runs only. CI actually *did run* `docker compose` in the PR, thanks to its' usage in the `$nobuild` branch below, and the runs are passing.

I believe `docker compose` works for me locally because I installed the plugin from the [docker pacakge source](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository). Also note that the Compose standalone tool is [not recommended](https://docs.docker.com/compose/install/#scenario-three-install-the-compose-standalone) and supported for compatibility purposes only. Do we know what is actually running in the CI?

If we think that it's not worth to spend time on another attempt figuring why `docker compose` is failing in the CI, we should revert all linux scripts to use `docker-compose` and document the debt, preferably in a GH issue.

Reply:
> The line this discussion refers to lives in the $buildonly branch which is currently used for local manual runs only. CI actually did run docker compose in the PR, thanks to its' usage in the $nobuild branch below, and the runs are passing.

CI does not use this script to run the test, only to build. It invokes docker-compose directly

https://github.com/dotnet/runtime/blob/4468ee51f5f13bd435e06a7c1084648af3211970/eng/pipelines/libraries/stress/http.yml#L65

> If we think that it's not worth to spend time on another attempt figuring out why docker compose is failing in the CI, we should revert all linux scripts to use docker-compose and document the debt, preferably in a GH issue.

Right, I did not notice that the `nobuild` branch still has the `docker compose` version. Will fix that.

Reply:
Ok, for some reason I mixed up the parameter names in my head 😅

Reply:
Ok, for some reason I mixed up the parameter names in my head 😅